buildroot-external/patches/occu/0151-WebUI-Add-DeviceRestoreConfigButton.patch

```diff
@@ -0,0 +1,121 @@
+--- occu/WebUI/www/config/ic_deviceparameters.cgi.orig
++++ occu/WebUI/www/config/ic_deviceparameters.cgi
+@@ -1133,6 +1133,29 @@
+   puts "</script>"
+ }
+ 
++proc get_user_easylinkmode {} {
++  global sid
++
++  set session [string trim $sid "@" ]
++  
++  set isecmd ""
++  append isecmd "var user=system.GetSessionVarStr('$session');"
++  array set user [rega_script $isecmd]
++  
++  set userid [lindex [split $user(user) ";"] 0]
++  
++  set isecmd ""
++  append isecmd "var easylinkmode = true;"
++  append isecmd "object user = dom.GetObject($userid);"
++  append isecmd "if (user)"
++  append isecmd "{"
++  append isecmd "easylinkmode = user.UserEasyLinkMode();"
++  append isecmd "}"
++  array set result [rega_script $isecmd]  
++  
++  return $result(easylinkmode);
++}
++
+ proc put_Header {} {
+ 
+   #cgi_debug -on
+@@ -1144,6 +1167,7 @@
+ 
+   set HmIPIdentifier "HmIP-RF"
+   set HmIPWIdentifier "HmIP-Wired"
++  set HMRFIdentifier "BidCos-RF"
+   set type $dev_descr(TYPE)
+ 
+   array set SENTRY ""
+@@ -1159,7 +1183,7 @@
+       set SENTRY(NAME) "$iface"
+       append SENTRY(NAME) ".$address"
+   }
+-
++  set EASYLINKMODE [get_user_easylinkmode]
+   set SENTRY(IMAGE) "<div id=\"picDiv_thumb\" class=\"CLASS22005\" onmouseover=\"picDivShow(jg_250, '$dev_descr(TYPE)', 250, '-1', this);\" onmouseout=\"picDivHide(jg_250);\"></div>"
+   append SENTRY(IMAGE) "<script type=\"text/javascript\">"
+   append SENTRY(IMAGE) "var jg_thumb = new jsGraphics(\"picDiv_thumb\");"
+@@ -1170,6 +1194,9 @@
+   set SENTRY(FIRMWARE) "<table id=\"id_firmware_table_$address\" class=\"id_firmware_table\" cellspacing=\"0\">"
+   #append SENTRY(FIRMWARE) "<tr><td>Version:</td><td class=\"CLASS22006\">$dev_descr(FIRMWARE)</td></tr>"
+   append SENTRY(FIRMWARE) "<tr><td>\${lblFirmwareVersion}</td><td id=\"curFwVersion\" class=\"CLASS22006\">$dev_descr(FIRMWARE)</td></tr>"
++  if {$EASYLINKMODE == "false" && $iface == $HMRFIdentifier} {
```

Comment:
Geht das "Restore Config" wirklich nur für BidCos-RF oder hast du das blos nicht für/mit homematicIP bis jetzt getestet?

Reply:
Da kein HmIP vorhanden, nicht getestet. Käme auf einen Versuch an

Reply:
Da DevConfig ja noch aus der Prä-IP Zeit stammt und vermutlich nie wirklich für IP angepasst wurde... lasst es lieber für IP-Geräte weg.

Begründung:
`Restoring config for 001E1D898A3967 failed`

![RM_SCI_nach_restore_Config](https://user-images.githubusercontent.com/54898249/141812763-a3235499-8ff9-4b51-9aa7-0494cfb46c59.JPG)

Sollte erstmal alc Contra-Argument reichen.

Reply:
> Restoring config for 001E1D898A3967 failed

Hast du den Config-Button am Gerät mal gedrückt?

Reply:
Du meinst die "Systemtaste"? 
Ja, die habe ich gedrückt. Dann ging die Servicemeldung für's Batt-Gerät (SCI) weg.
Man kann aber IP Geräte nicht wie HM Geräte durch Tastendruck in einen "warte auf Config Modus" versetzen... denke ich zumindest.

Reply:
>Man kann aber IP Geräte nicht wie HM Geräte durch Tastendruck in einen "warte auf Config Modus" versetzen... 

Macht auch nichts, das ist nicht Ziel des Button.

Ziel ist es, die komplette Config zur Übertragung anstehen zu lassen (was bei dir offensichtlich auch passiert ist).
Bei Netzgeräten geht sie halt sofort hin. 
Bei Batt.-Geräten erst durch drücken.